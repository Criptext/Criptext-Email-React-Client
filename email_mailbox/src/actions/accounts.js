import { Account } from './types';

export const addAccounts = accounts => {
  return {
    type: Account.ADD_BATCH,
    accounts
  };
};
