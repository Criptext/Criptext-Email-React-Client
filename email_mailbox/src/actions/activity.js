import { Activity } from './types';

export const logout = () => {
  return {
    type: Activity.LOGOUT
  };
};

export const stopAll = () => {
  return {
    type: Activity.STOP_ALL
  };
};

export const startLoadSync = () => {
  return {
    type: Activity.START_LOAD_SYNC
  };
};

export const stopLoadSync = () => {
  return {
    type: Activity.STOP_LOAD_SYNC
  };
};

export const startLoadThread = () => {
  return {
    type: Activity.START_LOAD_THREAD
  };
};

export const stopLoadThread = () => {
  return {
    type: Activity.STOP_LOAD_THREAD
  };
};

export const setAvatarUpdatedTimestamp = timestamp => {
  return {
    type: Activity.AVATAR_UPDATED_TIMESTAMP,
    timestamp
  };
};

export const updateLoadingSync = ({ totalTask, completedTask }) => {
  return {
    type: Activity.UPDATE_LOADING_SYNC,
    totalTask,
    completedTask
  };
};

export const updateSwitchThreads = ({ checked, disabled }) => {
  return {
    type: Activity.UPDATE_SWITCH_THREADS,
    checked,
    disabled
  };
};
