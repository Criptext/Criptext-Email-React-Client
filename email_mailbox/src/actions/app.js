import { App } from './types';
import {
  addAccounts,
  addContacts,
  addFeedItems,
  addLabels,
  addThreads,
  stopLoadThread
} from './index';
import { assembleAccounts } from './../utils/AccountUtils';
import { assembleLabels } from './../utils/LabelUtils';
import { assembleThreads } from './../utils/ThreadUtils';
import { assembleFeedItems } from './../utils/FeedItemUtils';
import { getGroupEvents } from './../utils/electronEventInterface';
const INIT_LIMIT_THREADS = 22;

export const addDataApp = ({
  account,
  activity,
  contact,
  email,
  file,
  feeditem,
  label,
  thread
}) => ({
  type: App.ADD_DATA,
  account,
  activity,
  contact,
  email,
  file,
  feeditem,
  label,
  thread
});

export const loadApp = params => {
  return async dispatch => {
    const accounts = await assembleAccounts();
    const labels = await assembleLabels();
    const { threads, contacts } = await assembleThreads({
      ...params,
      limit: INIT_LIMIT_THREADS
    });
    const feeditems = await assembleFeedItems();

    const account = addAccounts(accounts);
    const activity = stopLoadThread();
    const contact = addContacts(contacts);
    const feeditem = addFeedItems(feeditems, true);
    const label = addLabels(labels);
    const thread = addThreads(params.labelId, threads, true);

    dispatch(
      addDataApp({ account, activity, contact, label, feeditem, thread })
    );
    await getGroupEvents({});
  };
};
