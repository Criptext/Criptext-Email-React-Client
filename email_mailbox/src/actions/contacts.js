import { Contact } from './types';
import { defineContacts } from './../utils/ContactUtils';

export const addContacts = contacts => {
  return {
    type: Contact.ADD_BATCH,
    contacts
  };
};

export const loadContacts = ids => {
  return async dispatch => {
    try {
      const contacts = await defineContacts(ids);
      dispatch(addContacts(contacts));
    } catch (e) {
      // TO DO
    }
  };
};
