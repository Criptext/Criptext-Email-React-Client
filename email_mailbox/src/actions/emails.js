import { Email } from './types';
import {
  createEmailLabel,
  deleteEmailByKeys,
  deleteEmailContent,
  deleteEmailLabel,
  getContactByIds,
  getEmailsByIds,
  getEmailsByThreadId,
  getLabelsByText,
  postPeerEvent,
  unsendEmailEvent,
  updateEmail
} from '../utils/ipc';
import {
  addDataApp,
  addContacts,
  addFiles,
  unsendEmailFiles,
  updateEmailIdsThread,
  updateBadgeLabels,
  updateThreadsSuccess
} from './index';
import { EmailStatus, SocketCommand } from '../utils/const';
import {
  sendUnsendEmailErrorMessage,
  sendUnsendEmailExpiredErrorMessage,
  sendUpdateThreadLabelsErrorMessage,
  sendRemoveThreadsErrorMessage
} from './../utils/electronEventInterface';
import {
  filterCriptextRecipients,
  formEmailLabel
} from './../utils/EmailUtils';
import { defineContacts } from './../utils/ContactUtils';
import { defineFiles } from './../utils/FileUtils';

const eventlessEmailStatuses = [EmailStatus.FAIL, EmailStatus.SENDING];

export const addEmails = emails => {
  return {
    type: Email.ADD_BATCH,
    emails
  };
};

export const addEmailLabels = (emails, labelsAdd) => ({
  type: Email.ADD_LABEL,
  emails,
  labelsAdd
});

export const removeEmailLabels = (emails, labelsDelete) => ({
  type: Email.DELETE_LABEL,
  emails,
  labelsDelete
});

export const muteNotifications = emailId => {
  return {
    type: Email.MUTE,
    emailId
  };
};

export const markEmailUnreadSuccess = (emailId, unread) => {
  return {
    type: Email.MARK_UNREAD,
    emailId,
    unread
  };
};

export const loadEmails = ({ threadId, emailIds }) => {
  return async dispatch => {
    try {
      const data = await _loadEmails({ threadId, emailIds });
      const contactIds = Array.from(data.contactIds);
      const fileTokens = Array.from(data.fileTokens);

      const contacts = await defineContacts(contactIds);
      const contact = addContacts(contacts);
      const files = await defineFiles(fileTokens);
      const file = addFiles(files);
      const email = addEmails(data.emails);
      dispatch(addDataApp({ contact, email, file }));
    } catch (e) {
      // TO DO
    }
  };
};

export const muteEmail = (emailId, valueToSet) => {
  return async dispatch => {
    try {
      await updateEmail({ id: emailId, isMuted: valueToSet });
      dispatch(muteNotifications(emailId));
    } catch (e) {
      // To do
    }
  };
};

export const removeEmails = (labelId, emailsParams) => {
  return async dispatch => {
    try {
      const allMetadataKeys = emailsParams.map(param => param.key);
      const allEmailIds = emailsParams.map(param => param.id);
      const metadataKeys = emailsParams.reduce((result, item) => {
        if (!eventlessEmailStatuses.includes(item.status)) {
          result.push(item.key);
        }
        return result;
      }, []);

      const dbResponse = await deleteEmailByKeys(allMetadataKeys);

      if (dbResponse) {
        if (metadataKeys.length) {
          const eventParams = {
            cmd: SocketCommand.PEER_EMAIL_DELETED_PERMANENTLY,
            params: { metadataKeys }
          };
          const { status } = await postPeerEvent(eventParams);

          if (status === 200) {
            if (allMetadataKeys.length === 1) {
              const [email] = emailsParams;
              dispatch(
                updateEmailIdsThread({
                  labelId,
                  threadId: email.threadId,
                  emailIdToAdd: null,
                  emailIdsToRemove: [email.id]
                })
              );
            }

            dispatch(removeEmailsOnSuccess(allEmailIds));
          }
        }
      }
    } catch (e) {
      sendRemoveThreadsErrorMessage();
    }
  };
};

export const removeEmailsOnSuccess = emailIds => ({
  type: Email.REMOVE_EMAILS,
  emailIds
});

export const unsendEmail = params => {
  return async dispatch => {
    const { key, emailId, contactIds, unsentDate } = params;
    try {
      const contacts = await getContactByIds(contactIds);
      const emails = contacts.map(contact => contact.email);
      const criptextRecipients = filterCriptextRecipients(emails);
      const params = {
        metadataKey: Number(key),
        recipients: criptextRecipients
      };
      const { status } = await unsendEmailEvent(params);
      if (status === 200) {
        await updateEmail({
          key,
          status: EmailStatus.UNSEND,
          content: '',
          preview: '',
          unsentDate
        });
        await deleteEmailContent({ metadataKey: key });
        dispatch(unsendEmailFiles(emailId)).then(() =>
          dispatch(
            unsendEmailOnSuccess(
              String(emailId),
              unsentDate,
              EmailStatus.UNSEND
            )
          )
        );
      } else if (status === 405) {
        sendUnsendEmailExpiredErrorMessage();
      } else {
        sendUnsendEmailErrorMessage(status);
      }
    } catch (e) {
      sendUnsendEmailErrorMessage();
    }
  };
};

export const unsendEmailOnSuccess = (emailId, unsentDate, status) => ({
  type: Email.UNSEND,
  emailId,
  unsentDate,
  status
});

export const updateUnreadEmail = (
  labelId,
  threadId,
  emailId,
  key,
  valueToSet
) => {
  return async dispatch => {
    try {
      const metadataKeys = [key];
      const eventParams = {
        cmd: SocketCommand.PEER_EMAIL_READ_UPDATE,
        params: { metadataKeys, unread: valueToSet ? 1 : 0 }
      };
      const { status } = await postPeerEvent(eventParams);
      if (status === 200) {
        await updateEmail({ id: emailId, unread: !!valueToSet });
        dispatch(markEmailUnreadSuccess(emailId, valueToSet));
        dispatch(updateThreadsSuccess(labelId, [threadId], valueToSet));
        dispatch(updateBadgeLabels([labelId]));
      }
    } catch (e) {
      // To do
    }
  };
};

export const updateEmailOnSuccess = email => ({
  type: Email.UPDATE,
  email
});

export const updateEmailsSuccess = emails => ({
  type: Email.UPDATE_EMAILS,
  emails
});

export const updateEmailLabels = ({
  labelId,
  email,
  labelsAdded,
  labelsRemoved
}) => {
  return async dispatch => {
    try {
      if (email) {
        if (labelsAdded.length) {
          const addedLabels = await getLabelsByText(labelsAdded);
          const addedLabelsIds = addedLabels.map(label => label.id);
          const emailLabelsToAdd = formEmailLabel({
            emailId: email.id,
            labels: addedLabelsIds
          });

          await createEmailLabel(emailLabelsToAdd);
        }
        if (labelsRemoved.length) {
          const removedLabels = await getLabelsByText(labelsRemoved);
          const removedLabelsIds = removedLabels.map(label => label.id);
          const emailLabelsToRemove = {
            emailIds: [email.id],
            labelIds: removedLabelsIds
          };

          await deleteEmailLabel(emailLabelsToRemove);
        }

        let shouldDispatch;
        if (!eventlessEmailStatuses.includes(email.status)) {
          const eventParams = {
            cmd: SocketCommand.PEER_EMAIL_LABELS_UPDATE,
            params: {
              metadataKeys: [Number(email.key)],
              labelsAdded,
              labelsRemoved
            }
          };

          const { status } = await postPeerEvent(eventParams);

          shouldDispatch = status === 200;
        } else {
          shouldDispatch = true;
        }

        if (shouldDispatch) {
          dispatch(
            updateEmailIdsThread({
              labelId,
              threadId: email.threadId,
              emailIdToAdd: null,
              emailIdsToRemove: [email.id]
            })
          );
          if (labelsAdded.length) {
            const addedLabels = await getLabelsByText(labelsAdded);
            const addedLabelsIds = addedLabels.map(label => label.id);
            dispatch(addEmailLabels([email], addedLabelsIds));
          }
        } else {
          sendUpdateThreadLabelsErrorMessage();
        }
      }
    } catch (e) {
      sendUpdateThreadLabelsErrorMessage();
    }
  };
};

export const _loadEmails = async ({ threadId, emailIds }) => {
  const response = threadId
    ? await getEmailsByThreadId(threadId)
    : await getEmailsByIds(emailIds);
  return response.reduce(
    (result, element) => {
      element.fromContactIds = element.fromContactIds
        ? element.fromContactIds.split(',').map(Number)
        : [];
      element.to = element.to ? element.to.split(',').map(Number) : [];
      element.cc = element.cc ? element.cc.split(',').map(Number) : [];
      element.bcc = element.bcc ? element.bcc.split(',').map(Number) : [];
      const contactIds = [
        ...element.fromContactIds,
        ...element.to,
        ...element.cc,
        ...element.bcc
      ];
      const fileTokens = element.fileTokens
        ? element.fileTokens.split(',')
        : [];
      return {
        emails: { ...result.emails, [element.id]: element },
        fileTokens: new Set([...result.fileTokens, ...fileTokens]),
        contactIds: new Set([...result.contactIds, ...contactIds])
      };
    },
    { emails: {}, fileTokens: new Set(), contactIds: new Set() }
  );
};
