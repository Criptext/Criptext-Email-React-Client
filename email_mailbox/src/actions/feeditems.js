import { FeedItem } from './types';
import { deleteFeedItemById, updateFeedItem } from './../utils/ipc';
import { assembleFeedItems } from '../utils/FeedItemUtils';

export const addFeedItems = (feeds, clear) => {
  return {
    type: FeedItem.ADD_BATCH,
    feeds,
    clear
  };
};

export const removeFeedItemSuccess = feedItemId => {
  return {
    type: FeedItem.REMOVE_SUCCESS,
    feed: { id: feedItemId }
  };
};

export const updateFeedItemSuccess = ({ feedItemId, seen }) => {
  return {
    type: FeedItem.UPDATE_SUCCESS,
    feed: { id: feedItemId, seen }
  };
};

export const updateAllFeedItemsAsOlder = () => {
  return {
    type: FeedItem.UPDATE_ALL,
    field: 'isNew',
    value: false
  };
};

export const loadFeedItems = clear => {
  return async dispatch => {
    try {
      const feedItems = await assembleFeedItems();
      dispatch(addFeedItems(feedItems, clear));
    } catch (e) {
      // TO DO
    }
  };
};

export const selectFeedItem = feedItemId => {
  return async dispatch => {
    try {
      const seen = true;
      await updateFeedItem({ feedItemId, seen });
      dispatch(updateFeedItemSuccess({ feedItemId, seen }));
    } catch (e) {
      // TO DO
    }
  };
};

export const removeFeedItem = feedItemId => {
  return async dispatch => {
    try {
      await deleteFeedItemById(feedItemId);
      dispatch(removeFeedItemSuccess(feedItemId));
    } catch (e) {
      // TO DO
    }
  };
};
