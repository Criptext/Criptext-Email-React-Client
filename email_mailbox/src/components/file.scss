@import "./../styles/index.scss";

.file-container{
  margin: 5px 15px 5px 0;
  position: relative;

  .file-content{
    align-items: center;
    border: 1px solid transparent;
    border-radius: 6px;
    display: flex;
    height: 44px;
    padding: 7px 9px;
    position: relative;
    width: 278px;
  
    &:hover{
      cursor: pointer;

      .file-button-container{
        display: flex;
      }
    }
  
    &.file-loading {
      .file-content-icon, .file-content-detail{
        opacity: 0.5;
      }

      .file-button-cancel {
        display: flex;
      }
    }

    &.file-done {
      .file-content-icon{
        div{
          display: flex;
        }
      }
    }
  
    .file-content-icon{
      align-items: center;
      border-radius: 3px;
      display: flex;
      height: 30px;
      position: relative;
      width: 30px;

      &.file-content-icon-image { 
        background-color: #99ccff; 
      }

      &.file-content-icon-pdf { 
        background-color: #ff4539; 
      }

      &.file-content-icon-zip{ 
        background-color: #ffe196; 
      }

      &.file-content-icon-audio{ 
        background-color: #f0ba02;
      }

      &.file-content-icon-video{ 
        background-color: #00a49e;
      }

      &.file-content-icon-ppt{ 
        background-color: #d64123;
      }

      &.file-content-icon-word{ 
        background-color: #2b5697;
      }

      &.file-content-icon-excel{ 
        background-color: #4d9c45;
      }

      &.file-content-icon-file-default {
        background-color: #4a90e2; 
      }

      &.file-content-icon-unsent{
        > .unsent-icon {
          width: 30px;
          height: 30px;
          background-size: cover;
        }
      }

      >i{
        color: white;
        font-size: 26px;
        margin: 0 auto;
      }

      div.file-icon-status{
        align-items: center;
        background: rgb(0, 145, 255);
        border-radius: 50%;
        bottom: -5px;
        display: none;
        height: 11px;
        position: absolute;
        right: -5px;
        width: 11px;

        i{
          color: white;
          font-size: 15px;
          left: -2px;
          line-height: 15px;
          position: relative;
          top: 0px;
        }
      }
    }

    .file-content-detail{
      display: flex;
      flex-direction: column;
      margin-left: 10px;
      width: calc(100% - 32px);
    
      .file-detail-name{
        flex-grow: 1;
        font-size: 14px;
        font-weight: 700;
        max-height: 38px;
        overflow: hidden;
        padding-right: 5px;
        text-overflow: ellipsis;
        white-space: nowrap;
      }
    
      .file-detail-size{
        color: #9b9b9b;
        flex-grow: 0;
        font-size: 11px;
        white-space: nowrap;
      }
    }

    .file-button-container{
      align-items: center;
      border-radius: 6px;
      display: none;
      height: 58px;
      justify-content: flex-end;
      position: absolute;
      right: 0;
      top: 0;
      width: 70px;

      >button{
        background: transparent;
        margin-right: 10px;

        &.file-button-cancel{
          i{
            font-weight: bold;
          }
        }

        i{
          color: #89909a;
          font-size: 25px;
        }
      }
    }
  }

  .file-bar-loading{
    background-color: #0091ff;
    border-bottom-left-radius: 6px;
    border-bottom-right-radius: 6px;
    bottom: 1px;
    left: 2px;
    height: 2px;
    position: absolute;
  }
}

/* THEME
   ----------------------------- */
@mixin themable($_THEME, $_BACKGROUND_FILE, $_BORDER_COLOR_FILE, $_BACKGROUND_FILE_HOVER, $_COLOR_TEXT_FILE_TITLE, $_BACKGROUND_FILE_BUTTON, $_BACKGROUND_COLOR_FILE_UNAVAILABLE){
  [data-theme^=#{$_THEME}],
  [data-theme] [data-theme^=#{$_THEME}] {
    .file-container{
      .file-content{
        background-color: $_BACKGROUND_FILE;
        border-color: $_BORDER_COLOR_FILE;

        &:hover{
          background-color: $_BACKGROUND_FILE_HOVER;
        }

        .file-content-icon {
          &.file-content-icon-unsent{
            background-color: $_BACKGROUND_COLOR_FILE_UNAVAILABLE;
            > .unsent-icon {
              background-image: url(../#{$_PATH_TO_IMAGES}/file-unavailable-#{$_THEME}.svg);
            }
          }
        }

        .file-content-detail{
          .file-detail-name{
            color: $_COLOR_TEXT_FILE_TITLE;
          }
        }

        .file-button-container{
          background-image: $_BACKGROUND_FILE_BUTTON;
        }
      }
    }
  }
}

@include themable($_THEME_LIGHT, $_BACKGROUND_FILE_LIGHT, $_BORDER_COLOR_FILE_LIGHT, $_BACKGROUND_FILE_HOVER_LIGHT, $_COLOR_TEXT_FILE_TITLE_LIGHT, $_BACKGROUND_FILE_BUTTON_LIGHT, $_BACKGROUND_COLOR_FILE_UNAVAILABLE_LIGHT);
@include themable($_THEME_DARK, $_BACKGROUND_FILE_DARK, $_BORDER_COLOR_FILE_DARK, $_BACKGROUND_FILE_HOVER_DARK, $_COLOR_TEXT_FILE_TITLE_DARK, $_BACKGROUND_FILE_BUTTON_DARK, $_BACKGROUND_COLOR_FILE_UNAVAILABLE_DARK);