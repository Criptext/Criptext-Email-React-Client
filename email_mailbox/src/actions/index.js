import { addAccounts } from './accounts';
import {
  logout,
  setAvatarUpdatedTimestamp,
  startLoadSync,
  startLoadThread,
  stopAll,
  stopLoadSync,
  stopLoadThread,
  updateLoadingSync,
  updateSwitchThreads
} from './activity';
import { addContacts, loadContacts } from './contacts';
import { addFiles, loadFiles, unsendEmailFiles } from './files';
import {
  addLabelIdThread,
  addLabelIdThreadSuccess,
  addLabelIdThreadDraft,
  addLabelIdThreads,
  addLabelIdThreadsSuccess,
  addMoveLabelIdThreads,
  addThreads,
  filterThreadsOrLoadMoreByUnread,
  loadEvents,
  loadThreads,
  moveThreads,
  removeLabelIdThread,
  removeLabelIdThreadSuccess,
  removeLabelIdThreadDraft,
  removeLabelIdThreads,
  removeLabelIdThreadsSuccess,
  updateEmailIdsThread,
  updateThreadsSuccess,
  updateUnreadThreads,
  removeThreads,
  removeThreadsDrafts,
  removeThreadsSuccess,
  sendOpenEvent,
  updateThread
} from './threads';
import {
  addEmails,
  loadEmails,
  markEmailUnread,
  muteEmail,
  muteNotifications,
  removeEmails,
  removeEmailsOnSuccess,
  unsendEmail,
  unsendEmailOnSuccess,
  updateEmailLabels,
  updateEmailOnSuccess
} from './emails';
import {
  addLabels,
  addLabel,
  removeLabel,
  removeLabelOnSuccess,
  updateBadgeLabels,
  updateLabel,
  updateLabelSuccess
} from './labels';
import {
  addFeedItems,
  loadFeedItems,
  removeFeedItem,
  removeFeedItemSuccess,
  selectFeedItem,
  updateAllFeedItemsAsOlder,
  updateFeedItemSuccess
} from './feeditems';
import { loadSuggestions } from './suggestions';
import { addDataApp, loadApp } from './app';

export {
  addAccounts,
  addContacts,
  addDataApp,
  addEmails,
  addFiles,
  addFeedItems,
  addLabel,
  addLabels,
  addLabelIdThread,
  addLabelIdThreadSuccess,
  addLabelIdThreadDraft,
  addLabelIdThreads,
  addLabelIdThreadsSuccess,
  addMoveLabelIdThreads,
  addThreads,
  filterThreadsOrLoadMoreByUnread,
  loadApp,
  loadContacts,
  loadEmails,
  loadEvents,
  loadFiles,
  loadFeedItems,
  loadSuggestions,
  loadThreads,
  logout,
  markEmailUnread,
  moveThreads,
  muteEmail,
  muteNotifications,
  removeEmails,
  removeEmailsOnSuccess,
  removeLabel,
  removeLabelIdThread,
  removeLabelIdThreadSuccess,
  removeLabelIdThreadDraft,
  removeLabelIdThreads,
  removeLabelIdThreadsSuccess,
  removeLabelOnSuccess,
  removeFeedItem,
  removeFeedItemSuccess,
  removeThreads,
  removeThreadsDrafts,
  removeThreadsSuccess,
  selectFeedItem,
  sendOpenEvent,
  setAvatarUpdatedTimestamp,
  startLoadSync,
  startLoadThread,
  stopAll,
  stopLoadSync,
  stopLoadThread,
  unsendEmail,
  unsendEmailFiles,
  unsendEmailOnSuccess,
  updateAllFeedItemsAsOlder,
  updateBadgeLabels,
  updateFeedItemSuccess,
  updateEmailIdsThread,
  updateEmailLabels,
  updateEmailOnSuccess,
  updateLabel,
  updateLabelSuccess,
  updateLoadingSync,
  updateSwitchThreads,
  updateThread,
  updateThreadsSuccess,
  updateUnreadThreads
};
