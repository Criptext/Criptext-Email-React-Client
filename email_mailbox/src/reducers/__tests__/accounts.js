/* eslint-env node, jest */

import accountsReducer from './../accounts';
import * as actions from './../../actions/index';
import file from './../../../public/accounts.json';
const accounts = file.accounts;

jest.mock('./../../utils/ipc');
jest.mock('./../../utils/electronInterface');
jest.mock('./../../utils/electronEventInterface');

describe('Account actions - ADD_BATCH', () => {
  it('should add contacts', () => {
    const data = accounts.reduce(
      (result, element) => ({
        ...result,
        [element.id]: element
      }),
      {}
    );
    const action = actions.addAccounts(data);
    const state = accountsReducer(undefined, action);
    expect(state).toMatchSnapshot();
  });
});

describe('Account actions - UPDATE-ACCOUNTS', () => {
  it('should update: badge', () => {
    const data = [{ id: 2, badge: 1692 }];
    const action = actions.updateAccounts(data);
    const state = accountsReducer(undefined, action);
    expect(state).toMatchSnapshot();
  });
});
