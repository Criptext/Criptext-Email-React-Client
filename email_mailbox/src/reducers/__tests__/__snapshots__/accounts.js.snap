// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account actions - ADD_BATCH should add contacts 1`] = `
Immutable.Map {
  "1": Immutable.Map {
    "id": 1,
    "badge": undefined,
    "isActive": true,
    "name": "User test",
    "recipientId": "test",
  },
  "2": Immutable.Map {
    "id": 2,
    "badge": undefined,
    "isActive": false,
    "name": "User 2",
    "recipientId": "user2",
  },
  "3": Immutable.Map {
    "id": 3,
    "name": "User 3",
    "recipientId": "user3",
    "isActive": 0,
  },
  "4": Immutable.Map {
    "id": 4,
    "name": "User 4",
    "recipientId": "user4",
    "isActive": 0,
  },
}
`;

exports[`Account actions - UPDATE-ACCOUNTS should update: badge 1`] = `
Immutable.Map {
  "1": Immutable.Map {
    "id": 1,
    "badge": undefined,
    "isActive": true,
    "name": "User test",
    "recipientId": "test",
  },
  "2": Immutable.Map {
    "id": 2,
    "badge": 1692,
    "isActive": false,
    "name": "User 2",
    "recipientId": "user2",
  },
}
`;
